carp.studies.core/src/commonMain/kotlin/dk/cachet/carp/studies/application/StudyServiceHost.kt

```diff
@@ -257,6 +259,49 @@ class StudyServiceHost(
 
         repository.update( study )
 
-        return study.getStatus()
+        val participants = study.getParticipations( deploymentStatus.studyDeploymentId )
+        return ParticipantGroupStatus( deploymentStatus, participants )
+    }
+
+    /**
+     * Get the status of all deployed participant groups in the study with the specified [studyId].
+     *
+     * @throws IllegalArgumentException when a study with [studyId] does not exist.
+     */
+    override suspend fun getParticipantGroupStatusList( studyId: UUID ): List<ParticipantGroupStatus>
+    {
+        val study: Study? = repository.getById( studyId )
+        requireNotNull( study ) { "Study with the specified studyId is not found." }
+
+        // Get study deployment statuses.
+        // TODO: Participations are more logically stored per `studyDeploymentId`, rather than having to filter like this.
+        val studyDeploymentIds = study.participations.map { it.participation.studyDeploymentId }.toSet()
+        val studyDeploymentStatuses = deploymentService.getStudyDeploymentStatusList( studyDeploymentIds )
```

Comment:
Right now `DeanonymizedParticipation` (type of `study.participations`) stores within it `Participation`, which stores `studyDeploymentId`.  Wherever lists of `DeanonymizedParticipation`s are returned for the same deployment, there is thus a duplication of `studyDeploymentId`. Furthermore, as shown in this code, it makes much more sense to within study store participations in a map keyed on `studyDeploymentId`. I thus wanted to remove `Participation` from `DeanonymizedParticipation` and instead store `participationId`. This might impact the repo implementation, @ltj .

Reply:
I remember this being a bit weird but also very aligned with how participations are stored across deployments and studies. No problem with the proposed solution however